examples/Example21/electrons.cuh

```diff
@@ -0,0 +1,384 @@
+// SPDX-FileCopyrightText: 2022 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+#include "AdeptIntegration.cuh"
+
+#include <AdePT/BVHNavigator.h>
+#include <fieldPropagatorConstBz.h>
+
+#include <CopCore/PhysicalConstants.h>
+
+#define NOFLUCTUATION
+
+#include <G4HepEmElectronManager.hh>
+#include <G4HepEmElectronTrack.hh>
+#include <G4HepEmElectronInteractionBrem.hh>
+#include <G4HepEmElectronInteractionIoni.hh>
+#include <G4HepEmElectronInteractionUMSC.hh>
+#include <G4HepEmPositronInteractionAnnihilation.hh>
+// Pull in implementation.
+#include <G4HepEmRunUtils.icc>
+#include <G4HepEmInteractionUtils.icc>
+#include <G4HepEmElectronManager.icc>
+#include <G4HepEmElectronInteractionBrem.icc>
+#include <G4HepEmElectronInteractionIoni.icc>
+#include <G4HepEmElectronInteractionUMSC.icc>
+#include <G4HepEmPositronInteractionAnnihilation.icc>
+
+// Compute the physics and geometry step limit, transport the electrons while
+// applying the continuous effects and maybe a discrete process that could
+// generate secondaries.
+template <bool IsElectron, typename Scoring>
+static __device__ __forceinline__ void TransportElectrons(adept::TrackManager<Track> *electrons,
+                                                          Secondaries &secondaries, MParrayTracks *leakedQueue,
+                                                          Scoring *userScoring)
+{
+  using VolAuxData = AdeptIntegration::VolAuxData;
+#ifdef VECGEOM_FLOAT_PRECISION
+  const Precision kPush = 10 * vecgeom::kTolerance;
+#else
+  const Precision kPush = 0.;
+#endif
+  constexpr Precision kPushOutRegion = 10 * vecgeom::kTolerance;
+  constexpr int Charge               = IsElectron ? -1 : 1;
+  constexpr double Mass              = copcore::units::kElectronMassC2;
+  constexpr int Pdg                  = IsElectron ? 11 : -11;
+  fieldPropagatorConstBz fieldPropagatorBz(BzFieldValue);
+
+  int activeSize = electrons->fActiveTracks->size();
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < activeSize; i += blockDim.x * gridDim.x) {
+    const int slot      = (*electrons->fActiveTracks)[i];
+    Track &currentTrack = (*electrons)[slot];
+    auto energy         = currentTrack.energy;
+    auto pos            = currentTrack.pos;
+    auto dir            = currentTrack.dir;
+    auto navState       = currentTrack.navState;
+    const auto volume   = navState.Top();
+    adeptint::TrackData trackdata;
+    // the MCC vector is indexed by the logical volume id
+    const int lvolID          = volume->GetLogicalVolume()->id();
+    VolAuxData const &auxData = userScoring->GetAuxData_dev(lvolID);
+    assert(auxData.fGPUregion > 0); // make sure we don't get inconsistent region here
```

Comment:
I think we should remove this if we want at some point to allow region-based transport on top of this integration. Even if we can activate a region check on the host, we need to complement it with an exiting check on the device.

Reply:
removed